Sat Jun 17 03:53:40 CEST 2006  VMiklos <vmiklos@frugalware.org>
  * fwutil: descriptor leak fix in fwutil_init()
diff -rN -u old-frugalwareutils/libfwutil/libfwutil.c new-frugalwareutils/libfwutil/libfwutil.c
--- old-frugalwareutils/libfwutil/libfwutil.c	2006-06-17 14:00:25.000000000 +0200
+++ new-frugalwareutils/libfwutil/libfwutil.c	2006-06-17 14:00:25.000000000 +0200
@@ -150,7 +150,7 @@
 
 		if((fi = fopen("/proc/mounts", "r")))
 		{
-			if((fo = fopen("mtab", "w")))
+			if((fo = fopen("/etc/mtab", "w")))
 			{
 				char line[256];
 
@@ -161,7 +161,9 @@
 					if(!strstr(line, "root"))
 						fprintf(fo, "%s", line);
 				}
+				fclose(fo);
 			}
+			fclose(fi);
 		}
 
 		system("mount /dev/pts");

